packages/material-ui/src/ListItemButton/ListItemButton.js

```diff
@@ -0,0 +1,257 @@
+  };
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { alignItems, classes, dense, disabled, disableGutters, divider, selected } = styleProps;
+
+  const slots = {
+    root: [
+      'root',
+      dense && 'dense',
+      !disableGutters && 'gutters',
+      divider && 'divider',
+      disabled && 'disabled',
+      alignItems === 'flex-start' && 'alignItemsFlexStart',
+      selected && 'selected',
+    ],
+  };
+
+  const composedClasses = composeClasses(slots, getListItemButtonUtilityClass, classes);
+
+  return {
+    ...classes,
+    ...composedClasses,
+  };
+};
+
+const ListItemButtonRoot = styled(ButtonBase, {
+  shouldForwardProp: (prop) => rootShouldForwardProp(prop) || prop === 'classes',
+  name: 'MuiListItemButton',
+  slot: 'Root',
+  overridesResolver,
+})(({ theme, styleProps }) => ({
+  display: 'flex',
+  flexGrow: 1,
+  justifyContent: 'flex-start',
+  alignItems: 'center',
+  position: 'relative',
+  textDecoration: 'none',
+  boxSizing: 'border-box',
+  textAlign: 'left',
+  paddingTop: 8,
+  paddingBottom: 8,
+  transition: theme.transitions.create('background-color', {
+    duration: theme.transitions.duration.shortest,
+  }),
+  '&:hover': {
+    textDecoration: 'none',
+    backgroundColor: theme.palette.action.hover,
+    // Reset on touch devices, it doesn't add specificity
+    '@media (hover: none)': {
+      backgroundColor: 'transparent',
+    },
+  },
+  [`&.${listItemButtonClasses.selected}`]: {
+    backgroundColor: alpha(theme.palette.primary.main, theme.palette.action.selectedOpacity),
+    [`&.${listItemButtonClasses.focusVisible}`]: {
+      backgroundColor: alpha(
+        theme.palette.primary.main,
+        theme.palette.action.selectedOpacity + theme.palette.action.focusOpacity,
+      ),
+    },
+  },
+  [`&.${listItemButtonClasses.selected}:hover`]: {
+    backgroundColor: alpha(
+      theme.palette.primary.main,
+      theme.palette.action.selectedOpacity + theme.palette.action.hoverOpacity,
+    ),
+    // Reset on touch devices, it doesn't add specificity
+    '@media (hover: none)': {
+      backgroundColor: alpha(theme.palette.primary.main, theme.palette.action.selectedOpacity),
+    },
+  },
+  [`&.${listItemButtonClasses.focusVisible}`]: {
+    backgroundColor: theme.palette.action.focus,
+  },
+  [`&.${listItemButtonClasses.disabled}`]: {
+    opacity: theme.palette.action.disabledOpacity,
+  },
+  /* Styles applied to the root element if `divider={true}`. */
+  ...(styleProps.divider && {
+    borderBottom: `1px solid ${theme.palette.divider}`,
+    backgroundClip: 'padding-box',
+  }),
+  /* Styles applied to the root element if `alignItems="flex-start"`. */
+  ...(styleProps.alignItems === 'flex-start' && {
+    alignItems: 'flex-start',
+  }),
+  /* Styles applied to the root element unless `disableGutters={true}`. */
+  ...(!styleProps.disableGutters && {
+    paddingLeft: 16,
+    paddingRight: 16,
+  }),
+  /* Styles applied to the root element if dense. */
```

Comment:
No need of the comments for new code:

```suggestion
  ...(styleProps.divider && {
    borderBottom: `1px solid ${theme.palette.divider}`,
    backgroundClip: 'padding-box',
  }),
  ...(styleProps.alignItems === 'flex-start' && {
    alignItems: 'flex-start',
  }),
  ...(!styleProps.disableGutters && {
    paddingLeft: 16,
    paddingRight: 16,
  }),
```